go/pkg/log/server/property_test.go

```diff
@@ -86,99 +243,185 @@ func (suite *LogServerTestSuite) TestRecordLogDb_PushLogs() {
 				if int32(len(records)) != r.RecordCount {
 					t.Fatal("record count mismatch", len(records), r.RecordCount)
 				}
-				suite.model.CollectionData[c] = append(suite.model.CollectionData[c], records...)
 			},
-			"getAllCollectionsToCompact": func(t *rapid.T) {
+			"compaction": func(t *rapid.T) {
 				result, err := suite.logServer.GetAllCollectionInfoToCompact(ctx, &logservicepb.GetAllCollectionInfoToCompactRequest{})
 				assert.NoError(suite.t, err)
+
 				for _, collection := range result.AllCollectionInfo {
 					id, err := types.Parse(collection.CollectionId)
 					if err != nil {
 						t.Fatal(err)
 					}
-					compactionOffset := rapid.Int64Range(suite.model.CollectionCompactionOffset[id], int64(len(suite.model.CollectionData))).Draw(t, "new_position")
+					enumerationOffset := suite.model.CollectionEnumerationOffset[id]
+					compactionOffset := rapid.Uint64Range(suite.model.CollectionCompactionOffset[id], enumerationOffset).Draw(t, "new_position")
 					_, err = suite.logServer.UpdateCollectionLogOffset(ctx, &logservicepb.UpdateCollectionLogOffsetRequest{
 						CollectionId: id.String(),
-						LogOffset:    compactionOffset,
+						LogOffset:    int64(compactionOffset),
 					})
 					if err != nil {
 						t.Fatal(err)
 					}
 					suite.model.CollectionCompactionOffset[id] = compactionOffset
 				}
 			},
+			"getAllCollectionsToCompactWithMinCompactionSize": func(t *rapid.T) {
+				if len(suite.model.CollectionData) == 0 {
+					// Nothing to do if no data
+					return
+				}
+
+				// Determine the minimum compaction size by scanning over
+				// all the log data
+				minCompactionSize := uint64(math.MaxUint64)
```

Comment:
It would be good to encapsulate updates to the model.

Reply:
OK